src/helpers/validators.ts

```diff
@@ -222,3 +222,25 @@ export function asArrayOfDates(input: any): Date[] | undefined {
   if (dates.some((d: (Date | undefined)[]) => d === undefined)) throw new TypeError('Array contains invalid date');
   return dates;
 }
+
+/**
+ * Converts the inputs to a from and till date
+ * @param fromDate
+ * @param tillDate
+ * @throws Error - If tillDate is before fromDate
+ * @throws TypeError - If any of the inputs is not a valid date
+ */
+export function asFromAndTillDate(fromDate: any, tillDate: any): { fromDate: Date, tillDate: Date } {
+  const filters = {
+    fromDate: asDate(fromDate),
+    tillDate: asDate(tillDate),
+  };
+  filters.fromDate.setUTCHours(0, 0, 0, 0);
+  filters.tillDate.setUTCHours(0, 0, 0, 0);
```

Comment:
I think we should really prefer that people only payout / report full days, as otherwise things will go messy

I am not sure how to do this though in a nicer way with timestamps.

Reply:
So I looked more into it, and I do agree with your that it would be nice to adhere to the timestamp as give, but that does mean we should make the endDate inclusive.

If you have a seller payout with a endDate which is exactly a transaction then it wont fall within the report, which would be weird, but making the endDate inclusive also feels weird.

Reply:
I think it is best to have the endDate be exclusive and the startDate be inclusive. This is currently the standard across SudoSOS, also when you use time ranges as filters for your query. As long as we are consistent, I think it is fine.

Reply:
The issue with this is that when I later, in the seller payouts pdf, get the transactions in the range of the start and end date it would miss some transactions right around the edge. See the todo in my code here: https://github.com/GEWIS/sudosos-backend/blob/1c3c2bcfe0d22ec0f5e5839f3d4f13bc183ee9ae/src/service/pdf/seller-payout-pdf-service.ts#L40

But I am not sure what to do, its just that if we use _exact_ time stamps very close the the actual transactions the report is about, the chance is high that it will miss something.

Reply:
This is indeed possible, but I think it is not up to us to implement all those restrictions just because you _could_ miss a transaction. Regarding SellerPayouts, I think it is better to have other failsafes in place, like the user's balance (which should be zero).

Oh and about the PDF, I think we should *not* throw an error when the amount does not match. Fetching the transaction report is the only way to get the new amount if history has been changed. It also allows you to find discrepancies between the actual amount and the amount set in SudoSOS (because they will not match on the PDF, but they should!)

Reply:
> Oh and about the PDF, I think we should not throw an error when the amount does not match. Fetching the transaction report is the only way to get the new amount if history has been changed. It also allows you to find discrepancies between the actual amount and the amount set in SudoSOS (because they will not match on the PDF, but they should!)

But in this case, the PDF would be wrong.